checkout

```diff
@@ -0,0 +1,155 @@
+    echo
+    echo "       <tag-or-branch> Name of the tag or branch to check out."
+    echo "                       BRANCH: Check out the branch in scripts, coreos-overlay"
+    echo "                                and portage-stable, and pull / rebase to the"
+    echo "                                latest changes by default (but see '-n' below)."
+    echo "                       TAG:    Check out the tag in scripts, and update submodules."
+    echo
+    echo "       -t            force TAG mode. Branches of the given name are ignored."
+    echo "                      Mutually exclusive to '-b'."
+    echo "       -b            force BRANCH mode. Tags of the given name are ignored."
+    echo "                      Mutually exclusive to '-t'."
+    echo
+    echo "       -n            When checking out branches, omit 'git pull --rebase'."
+    echo
+    echo "       -h            Print this help."
+    echo
+}
+# --
+
+while [ 0 -lt $# ] ; do
+    case "$1" in
+    -h) usage; exit 0;;
+    -t) force_tag="true";       shift;;
+    -b) force_branch="true";    shift;;
+    -n) update_branch="false";  shift;;
+    *) if [ -n "$name" ] ; then
+            echo
+            echo "ERROR: only ONE tag-or-branch can be specified."
+            echo
+            usage
+            exit 1
+       fi
+       name="$1";      shift;;
+    esac
+done
+
+if [ -z "$name" ] ; then
+    echo
+    echo "ERROR: <tag-or-branch> is mandatory."
+    echo
+    usage
+    exit 1
+fi
+
+if $force_branch && $force_tag; then
+    echo
+    echo "ERROR: '-t' and '-b' are mutually exclusive. Please make up your mind."
+    echo
+    usage
+    exit 1
+fi
+# --
+
+# make sure submodules are initialised
+git submodule init
+for dir in sdk_container/src/third_party/coreos-overlay \
+           sdk_container/src/third_party/portage-stable ; do
+    if [ ! -f "$dir"/.git ] ; then
+        git submodule update -N "$dir"
+    fi
+done
+
+function check_all() {
+    local gitcmd="$1"
+    local name="$2"
+
+    local scripts="$(git $gitcmd \
+                        | sed -e 's/^[[:space:]]*//' -e 's:remotes/[^/]\+/::' \
+                        | grep -m1 -E "^$name\$")"
+    local overlay="$(git -C sdk_container/src/third_party/coreos-overlay $gitcmd \
+                        | sed -e 's/^[[:space:]]*//' -e 's:remotes/[^/]\+/::' \
+                        | grep -m1 -E "^$name\$")"
+    local portage="$(git -C sdk_container/src/third_party/portage-stable $gitcmd \
+                        | sed -e 's/^[[:space:]]*//' -e 's:remotes/[^/]\+/::' \
+                        | grep -m1 -E "^$name\$")"
+
+    if [ -n "$scripts" -a -n "$overlay" -a -n "$portage" ] ; then
+        echo "$scripts"
+    fi
+}
+# --
+
+#
+# TAG
+#
+if ! $force_branch; then 
+    for dir in . \
+        sdk_container/src/third_party/coreos-overlay \
+        sdk_container/src/third_party/portage-stable ; do
+        git -C "$dir" fetch --tags
```

Comment:
This fails as soon as a retag was done by someone, I think it's better to force the overwriting of local tags instead of bailing out

Reply:
That's a good catch, will update.